### PR TITLE
Streamline protocol layer classes, issue #67

### PR DESCRIPTION
**Pull request by ptesavol**
Implemented EndpointListener interface as a Javascript class, and made all protocol classes to implement this interface. This removes duplicated code (importing and binding Endpoint events) from the protocol classes, and makes sure that the protocol classes listen to all the events coming from the lower layer.

The "implement(implementor, endpoint)" method of EndpointListener class checks whether the implementor has actually implemented the interface's methods and throws an exception otherwise. If the methods exist as claimed, it binds the corresponding endpoint events to the methods.

Jslint config was slightly modified to only give a warning when empty placeholder methods are encountered. I also removed the warning about exceeding the line length, which happens in almost all the files in the project.

### CODE REVIEW

**Comment by mirpo** [comment]
what do you think about https://www.npmjs.com/package/implement-js ?

**Review by mirpo — APPROVED** [review]

**Comment by ptesavol** [comment]
Implement-js looks promising in the sense that it is run-time (as oppesed to flowtype). However there seems to be an open issue concerning es6 classes: https://github.com/Jahans3/Implement.js/issues/6

File: .eslintrc.js [hunk 1]
```diff
@@ -56,5 +56,8 @@ module.exports = exports = {
             numbers: true
         }],
         'semi': [ERROR, 'never'],
+        'max-len': DISABLED,
```
Comment (harbu) [inline .eslintrc.js]: Can we still leave this at `WARN`? We have it currently set up at 150 char limit, and that is already quite a long line.

**Review by harbu — CHANGES_REQUESTED** [review]
Looks good to me, but I think we should leave the max-len to WARN.

File: .eslintrc.js (hunk 1 above)
Comment (ptesavol) [reply .eslintrc.js]: yep, I'll make the change!

**Review by harbu — APPROVED** [review]

**Pull request merged**